### PR TITLE
Switch integration test to manually starting kube & openshift controller managers

### PR DESCRIPTION
**Pull request by soltysh**
/assign @deads2k

### CODE REVIEW

File: vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go [hunk 1]
```diff
@@ -106,6 +106,27 @@ controller, and serviceaccounts controller.`,
 				os.Exit(1)
 			}
 		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			verflag.PrintAndExitIfRequested()
+			utilflag.PrintFlags(cmd.Flags())
+
+			c, err := s.Config(KnownControllers(), ControllersDisabledByDefault.List())
+			if err != nil {
+				return err
```
Comment (deads2k) [inline vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go]: Can't we keep the exits?  Is it always a problem or just when there's a bug in teh binary?

Comment (soltysh) [reply vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go]: although that's equal with `glog.Fatal` so maybe we should be good with it

**Comment by deads2k** [comment]
if it works

/lgtm

File: test/util/server/server.go [hunk 2]
```diff
@@ -356,17 +358,11 @@ func StartConfiguredMasterWithOptions(masterConfig *configapi.MasterConfig) (str
...
-	/********** controllers the old way **********/
-
-	// controllers need new address they can bind debug and healthz to
-	controllersAddrStr, err := FindAvailableBindAddress(10000, 29999)
-	if err != nil {
-		return "", fmt.Errorf("Couldn't find free address for controllers: %v", err)
+	if err := startKubernetesControllers(masterConfig, adminKubeConfigFile); err != nil {
+		return "", err
 	}
-	masterConfig.ServingInfo.BindAddress = controllersAddrStr
```
Comment (soltysh) [inline test/util/server/server.go]: @deads2k ha, this switch uncovered an error, that we're serving more ciphers in the API that we should. The `TestMasterTLSDefaults` test before was working against controller address and that's why it passed, but should fail :man_facepalming: another :wilted_flower:

**Comment by soltysh** [comment]
@deads2k I've bumped the timeout for `TestClusterQuota` and most importantly fixed that cipher thing, not sure if that's where you wanted it to be? ptal

**Comment by soltysh** [comment]
That cluster quota test will require additional love, it times out too often :/ 
/retest

**Comment by soltysh** [comment]
Ok, so that TestClusterQuota failure looks real, looking into it.

File: pkg/configconversion/simple.go [hunk 3]
```diff
@@ -14,6 +15,9 @@ func ToHTTPServingInfo(in *legacyconfigv1.HTTPServingInfo) (out configv1.HTTPSer
 	if err != nil {
 		return configv1.HTTPServingInfo{}, err
 	}
+	if len(out.CipherSuites) == 0 {
```
Comment (deads2k) [inline pkg/configconversion/simple.go]: This is ultimately dead code, but I am surprised that https://github.com/openshift/library-go/blob/master/pkg/config/configdefaults/config_default.go#L53-L55 doesn't cover it.

File: test/integration/clusterquota_test.go [hunk 4]
```diff
@@ -189,5 +190,10 @@ func waitForQuotaStatus(clusterAdminClient quotaclient.Interface, name string, c
 		}
 		return false, nil
 	})
-
+	if err == nil {
+		// since now we run each process separately we need to wait for the informers
+		// to catch up on the update and only then continue
+		time.Sleep(3 * time.Second)
```
Comment (deads2k) [inline test/integration/clusterquota_test.go]: omfg.  The fact that I'll take this is a sign of how desperate I am.  Make an issue on our jira board to remove this somehow.

**Comment by deads2k** [comment]
It's pretty ugly, but I think we must push through it.  If it's green.

/lgtm

File: test/integration/clusterquota_test.go (hunk 4 above)
Comment (soltysh) [reply test/integration/clusterquota_test.go]: https://jira.coreos.com/browse/MSTR-222

File: pkg/configconversion/simple.go (hunk 3 above)
Comment (soltysh) [reply pkg/configconversion/simple.go]: I think once we switch integration to properly use the new configs then yeah it should be good. But for now I'm leaving it as is. I'll open a JIRA for that: https://jira.coreos.com/browse/MSTR-223

**Pull request merged**